ion/unmarshal_test.go

```diff
@@ -340,11 +340,11 @@ func TestUnmarshalBinary(t *testing.T) {
 
 	var symbolVal string
 	symbolBytes := prefixIVM([]byte{0x71, 0x09}) // $9
-	test(symbolBytes, symbolVal, "$ion_shared_symbol_table")
+	test(symbolBytes, symbolVal, newString("$ion_shared_symbol_table"))
 
 	var stringVal string
 	stringBytes := prefixIVM([]byte{0x83, 'a', 'b', 'c'}) // "abc"
-	test(stringBytes, stringVal, "abc")
+	test(stringBytes, stringVal, newString("abc"))
```

Comment:
I'm not sure I understand the value of [the `newString` function](https://github.com/amzn/ion-go/blob/58d1539a051c268383698a3566e3ef176d994c1e/ion/symboltoken_test.go#L25-L27):

```go
func newString(value string) *string {
	return &value
}
```

It's certainly correct, but why is `newString("hello")` preferable to `&"hello"`?

Reply:
Because only one of them successfully compiles? :) It’s a bit of a chicken-and-egg: it’s not very “go-ey” to use pointers to basic types in APIs, so there isn’t a lot of built-in syntax sugar to make working with them easy (in this case: you can’t directly take the address of a literal, you need to assign it to a variable and then take the address of the variable), which reinforces the convention against using them in APIs.

Reply:
> Because only one of them successfully compiles? :)

That's a good reason. :)

> you can’t directly take the address of a literal, you need to assign it to a variable and then take the address of the variable

Huh.

Reply:
This isn't a choice between `newString("hello")` and `&"hello"`. 
Rather it's a choice between `newString("hello")` and 
```go
str := "hello"
&str
```

Reply:
I understood what @fernomac was describing, I was just surprised to learn that that was a restriction in Go.